### PR TITLE
deal-pipeline-permission

### PR DESCRIPTION
**Pull request by Enkhtuvshin0513**

### CODE REVIEW

File: src/__tests__/dealMutations.test.ts [hunk 1]
```diff
@@ -113,6 +117,7 @@ describe('Test deals mutations', () => {
       name: 'deal pipeline',
       boardId: board._id,
       stages: [stage.toJSON()],
+      type: 'public',
```
Comment (batamar) [inline src/__tests__/dealMutations.test.ts]: use the meaningful variable name. Rename to scope

File: src/db/models/definitions/constants.ts [hunk 2]
```diff
@@ -214,6 +214,12 @@ export const PRODUCT_TYPES = {
   ALL: ['product', 'service'],
 };
 
+export const PIPELINE_TYPES = {
```
Comment (batamar) [inline src/db/models/definitions/constants.ts]: rename accordingly

File: src/data/resolvers/queries/deals.ts [hunk 3]
```diff
@@ -48,17 +48,26 @@ const dealQueries = {
...
+      return DealPipelines.find({ boardId }).sort({ order: 1, createdAt: -1 });
+    }
 
+    return DealPipelines.find({
+      $and: [
+        { boardId },
+        { $or: [{ type: 'public' }, { type: { $exists: false } }, { memberIds: user._id }, { userId: user._id }] },
+      ],
+    }).sort({ order: 1, createdAt: -1 });
+  },
```
Comment (munkhjin0223) [inline src/data/resolvers/queries/deals.ts]: Separate into two different functions and refactor this function. And add migrate.

File: src/data/schema/deal.ts [hunk 4]
```diff
@@ -15,6 +15,8 @@ export const types = `
     _id: String!
     name: String!
     boardId: String!
+    type: String
```
Comment (munkhjin0223) [inline src/data/schema/deal.ts]: type!: String

**Pull request merged**